// @fortawesome/angular-fontawesome/fesm2015/fortawesome-angular-fontawesome-testing.js
import { ɵɵdefineInjectable, Injectable, NgModule } from '@angular/core';
import { FontAwesomeModule, FaIconLibrary } from '@fortawesome/angular-fontawesome';

import * as ɵngcc0 from '@angular/core';
const dummyIcon = {
    prefix: 'fad',
    iconName: 'dummy',
    icon: [512, 512, [], 'f030', 'M50 50 H462 V462 H50 Z'],
};
class MockFaIconLibrary {
    addIcons() {
        throw new Error('Attempt to add an icon to the MockFaIconLibrary.');
    }
    addIconPacks() {
        throw new Error('Attempt to add an icon pack to the MockFaIconLibrary.');
    }
    getIconDefinition(prefix, name) {
        return dummyIcon;
    }
}
MockFaIconLibrary.ngInjectableDef = ɵngcc0.ɵɵdefineInjectable({ token: MockFaIconLibrary, factory: function MockFaIconLibrary_Factory(t) { return new (t || MockFaIconLibrary)(); }, providedIn: 'root' });
/*@__PURE__*/ ɵngcc0.ɵsetClassMetadata(MockFaIconLibrary, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], null, { addIcons: [], addIconPacks: [], getIconDefinition: [] });
MockFaIconLibrary.ɵprov = ɵɵdefineInjectable({ factory: function MockFaIconLibrary_Factory() { return new MockFaIconLibrary(); }, token: MockFaIconLibrary, providedIn: "root" });

class FontAwesomeTestingModule {
}
FontAwesomeTestingModule.ngModuleDef = ɵngcc0.ɵɵdefineNgModule({ type: FontAwesomeTestingModule });
/*@__PURE__*/ ɵngcc0.ɵɵsetNgModuleScope(FontAwesomeTestingModule, { exports: function () { return [FontAwesomeModule]; } });
/*@__PURE__*/ ɵngcc0.ɵsetClassMetadata(FontAwesomeTestingModule, [{
        type: NgModule,
        args: [{
                exports: [FontAwesomeModule],
                providers: [{ provide: FaIconLibrary, useExisting: MockFaIconLibrary }]
            }]
    }], null, null);
FontAwesomeTestingModule.ngInjectorDef = ɵngcc0.ɵɵdefineInjector({ factory: function FontAwesomeTestingModule_Factory(t) { return new (t || FontAwesomeTestingModule)(); }, providers: [{ provide: FaIconLibrary, useExisting: MockFaIconLibrary }], imports: [FontAwesomeModule] });

/**
 * Generated bundle index. Do not edit.
 */

export { FontAwesomeTestingModule, MockFaIconLibrary as ɵa };
