// @fortawesome/angular-fontawesome/bundles/fortawesome-angular-fontawesome-testing.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@fortawesome/angular-fontawesome'),require('@angular/core'),require('@fortawesome/angular-fontawesome')) :
    typeof define === 'function' && define.amd ? define('@fortawesome/angular-fontawesome/testing', ['exports', '@angular/core', '@fortawesome/angular-fontawesome','@angular/core','@fortawesome/angular-fontawesome'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.fortawesome = global.fortawesome || {}, global.fortawesome['angular-fontawesome'] = global.fortawesome['angular-fontawesome'] || {}, global.fortawesome['angular-fontawesome'].testing = {}), global.ng.core, global.fortawesome['angular-fontawesome']));
}(this, (function (exports, i0, angularFontawesome,ɵngcc0,ɵngcc1) { 'use strict';

    var dummyIcon = {
        prefix: 'fad',
        iconName: 'dummy',
        icon: [512, 512, [], 'f030', 'M50 50 H462 V462 H50 Z'],
    };
    var MockFaIconLibrary = /** @class */ (function () {
        function MockFaIconLibrary() {
        }
        MockFaIconLibrary.prototype.addIcons = function () {
            throw new Error('Attempt to add an icon to the MockFaIconLibrary.');
        };
        MockFaIconLibrary.prototype.addIconPacks = function () {
            throw new Error('Attempt to add an icon pack to the MockFaIconLibrary.');
        };
        MockFaIconLibrary.prototype.getIconDefinition = function (prefix, name) {
            return dummyIcon;
        };
MockFaIconLibrary.ngInjectableDef = ɵngcc0.ɵɵdefineInjectable({ token: MockFaIconLibrary, factory: function MockFaIconLibrary_Factory(t) { return new (t || MockFaIconLibrary)(); }, providedIn: 'root' });
/*@__PURE__*/ ɵngcc0.ɵsetClassMetadata(MockFaIconLibrary, [{
        type: i0.Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, { addIcons: [], addIconPacks: [], getIconDefinition: [] });
        return MockFaIconLibrary;
    }());
    MockFaIconLibrary.ɵprov = i0.ɵɵdefineInjectable({ factory: function MockFaIconLibrary_Factory() { return new MockFaIconLibrary(); }, token: MockFaIconLibrary, providedIn: "root" });

    var FontAwesomeTestingModule = /** @class */ (function () {
        function FontAwesomeTestingModule() {
        }
FontAwesomeTestingModule.ngModuleDef = ɵngcc0.ɵɵdefineNgModule({ type: FontAwesomeTestingModule });
/*@__PURE__*/ ɵngcc0.ɵɵsetNgModuleScope(FontAwesomeTestingModule, { exports: [ɵngcc1.FontAwesomeModule] });
/*@__PURE__*/ ɵngcc0.ɵsetClassMetadata(FontAwesomeTestingModule, [{
        type: i0.NgModule,
        args: [{
                exports: [angularFontawesome.FontAwesomeModule],
                providers: [{ provide: angularFontawesome.FaIconLibrary, useExisting: MockFaIconLibrary }]
            }]
    }], function () { return []; }, null);
FontAwesomeTestingModule.ngInjectorDef = ɵngcc0.ɵɵdefineInjector({ factory: function FontAwesomeTestingModule_Factory(t) { return new (t || FontAwesomeTestingModule)(); }, providers: [{ provide: angularFontawesome.FaIconLibrary, useExisting: MockFaIconLibrary }], imports: [ɵngcc1.FontAwesomeModule] });
        return FontAwesomeTestingModule;
    }());

    /**
     * Generated bundle index. Do not edit.
     */

    exports.FontAwesomeTestingModule = FontAwesomeTestingModule;
    exports.ɵa = MockFaIconLibrary;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
